src/store/index-level.ts

```diff
@@ -0,0 +1,205 @@
+import type { LevelDatabase } from './create-level.js';
+import type { AbstractBatchDelOperation, AbstractBatchOperation, AbstractIteratorOptions } from 'abstract-level';
+
+import { abortOr } from '../utils/abort.js';
+import { createLevelDatabase } from './create-level.js';
+import flat from 'flat';
+
+const KEY_DELIMITER = '<';
+const KEY_BOUNDARY = '>';
+
+const VALUE_DELIMITER = '=';
+
+const KEY_PREFIX_DELIMITER = '@';
```

Comment:
i was trying to find things that are unlikely to be used in keys or generated by base64 things or etc.

i think a better approach might actually be to leverage `JSON.stringify` to guarantee no chance of collision

Reply:
+1 for using less likely characters

Reply:
so i thought about this more and `JSON.stringify` actually wouldn't really work as well cause it could cause issues with searching for something like `gt: 'a'` with a value of `'abcd'` since stringifying it would result in `'a'`, meaning that it would no longer lexicographically compare the same :(

Reply:
i did a bit more playing around with this and i was able to replace the while `KEY_BOUNDARY` by just adding a `findExactMatch` since ultimately that was the only part we were missing

Reply:
so i did even more exploration and i dont think there's a good way around this (at least not in JS)
- i tried leveraging `sublevel` to do the partitioning (in the hopes that it would be smart about encoding things to prevent reserved characters) but apparently `sublevel` restricts keys to be between the one after the reserved character (which defaults to `!`/`\x21`) and `\x1F`, so we'd be unable to use unicode in any value of anything, which is probably not ideal (e.g. querying for a user's name in countries that dont use ASCII)
- i tried using `JSON.stringify` to encode everything but that would affect the lexicographic order of the values, both in what i mentioned above and it transforming unicode (e.g. `\x00` becomes `\\u0000`, which would now compare differently if searching for `\\`)

i think we probably just have to hope that nobody will use any of the reserved characters (which is why i changed them to `\x01` and `\x02`) or other screwy things

Reply:
Thanks for the due diligence! On board with not using common chars, and you've gone above and beyond at this point, so nothing else to say from my side on this!